lib/pull.ml

```diff
@@ -109,12 +125,24 @@ let write_duniverse_dir_documentation ~duniverse_dir =
   in
   written
 
-let duniverse ~full ~root ~global_state ~trim_clone duniverse =
+let filter_preserved ~preserved duniverse =
+  List.filter_map
```

Comment:
I think this could be simplified to
```ocaml
List.filter ~f:(fun entry : Duniverse.Repo.t -> not (List.mem entry.dir ~set:preserved)) duniverse
```
But I get if you think it's easier to read in its current form.

Reply:
True, good catch. There is a point about readability but there's another point about not overcomplicating it and I did this accidentally. Thanks for pointing it out, I missed the forest for the trees.